src/WinRT.Runtime/ComWrappersSupport.cs

```diff
@@ -68,34 +68,44 @@ public static void MarshalDelegateInvoke<T>(IntPtr thisPtr, Action<T> invoke)
 
         // If we are free threaded, we do not need to keep track of context.
         // This can either be if the object implements IAgileObject or the free threaded marshaler.
-        internal unsafe static bool IsFreeThreaded(IObjectReference objRef)
+        internal unsafe static bool IsFreeThreaded(IntPtr iUnknown)
         {
-            if (objRef.TryAs(InterfaceIIDs.IAgileObject_IID, out var agilePtr) >= 0)
+            Guid iid_IAgileObject = InterfaceIIDs.IAgileObject_IID;
+            if (Marshal.QueryInterface(iUnknown, ref iid_IAgileObject, out var agilePtr) >= 0)

```

Comment:
Can skip the temp copy here (and in all other equivalent cases):
```suggestion
            if (Marshal.QueryInterface(iUnknown, ref Unsafe.AsRef(InterfaceIIDs.IAgileObject_IID), out var agilePtr) >= 0)
```

Reply:
Will do a pass.